torch/quantization/convert_modules.py

```diff
@@ -0,0 +1,221 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import torch.nn as nn
+import torch.nn.quantized as nnq
+import torch
+from .observer import *
+
+def _forward_hook(self, input, output):
+    for module in self.children():
+        module.forward(output)
+
+def addObserver(myMod, qConfigDict, qConfigParent=None, prefix=''):
+    r"""Transform a module to a quantized module according to qConfigDict
+
+    This function insert observer module to all leaf child module of a given
+    module based on qConfigDict.
+
+    Args:
+        myMod: instance of the module we want to transform
+        qCofnigDict: dictionary that maps from name of submodule to quantization
+                     configuration
+        qConfigParent: quantization config of parent module, we will fallback to
+                       this config when there is no specified config for current
+                       module
+        prefix: corresponding prefix of the current module, used as key in
+                qConfigDict
+    """
+    myMod.qConfig = None
+    if prefix in qConfigDict:
+        myMod.qConfig = qConfigDict[prefix]
+    else:
+        if qConfigParent is not None:
+            myMod.qConfig = qConfigParent
+    # Insert observers only for leaf nodes
+    if myMod.qConfig is not None and len(myMod._modules) == 0:
+        myMod.add_module('observer', myMod.qConfig.activation())
+        myMod.register_forward_hook(_forward_hook)
+
+    for name, child in myMod.named_children():
+        if name is not 'observer':
```

Comment:
you add observer only if there are no children, so you can remove it and just put the whole loop into for

also `is not 'observer'` doesn't do what you think it does: https://stackoverflow.com/questions/2988017/string-comparison-in-python-is-vs

Reply:
I think the reason is adding observer changed 'myMod.named_children()' it will go into infinite loop without this check..